### ISSUE TITLE
Feature Wish List

### ISSUE THREAD

**Issue by officert** [officert/mongotron#46]
Let us know what features you'd like to see in Mongotron:

**Comment by officert** [comment]
Importing and exporting connection settings.

Connections settings are a pain to put in, so it'd be nice if you could export your connections as a backup, and import them too.

**Comment by kinduff** [comment]
The application should keep configuration and content in separated views. Settings currently uses the tab space to show its content, my thoughts are that all app configuration related settings should have an independent space.

**Comment by ah-** [comment]
Full support for the different auth mechanisms! Mongotron looks amazing, I'd love to be able to use it with all servers, not just open ones.

**Comment by jojobyte** [comment]
Dear Santa, 

- I would like SSH Tunnel Support.
- Inline Editing, cause I'm lazy, or at least auto-fill the query field with the desired item to edit... cause I'm lazy.

**Comment by officert** [comment]
@kinduff I agree with getting the settings stuff out of a tab. If you have any ideas on a better place to put them I'd love to hear. I had thought to put them in a modal, like the connection manager, but it seems like too much stuff for a single modal.

**Comment by officert** [comment]
@jojobyte I created a new issue for adding SSH support.

https://github.com/officert/mongotron/issues/56

I'm not entirely sure what is required for this, if you have any more details feel free to fill in the issue with them.

@ah- Are there other auth mechanisms you'd like to see?

**Comment by officert** [comment]
@jojobyte for editing documents I started to play around with editing in a modal. You would right-click on a doc and see a context-menu with an option to 'Edit Document'. This would bring up a modal where you could make your edits and save.

But I am also open to editing directly inline if we can figure out a good way to do it. I think we should start with the modal, as it's a bit easier to implement IMO, and then explore how we might add inline editing capabilities as well.

**Comment by jojobyte** [comment]
@kinduff & @officert I sort of disagree about settings. As this is built in electron its semi-standard practice as far as I can tell to do it in a tab. That's how Atom does it and every other electron app I've tried.

@officert Sounds good. I was thinking just a little pencil edit icon next to the delete one in the Code View, which could just as easily trigger a modal as inline editing. So Long as I don't have to copy & paste the object to edit it I'll be happy.

**Comment by ah-** [comment]
@officert X509 certificate authentication would be really useful.

**Comment by kinduff** [comment]
@jojobyte Well, since it's a semi-standard practice, we should be able to propose a better solution. My suggestion is to keep them separated with the same argument as before, a modal (?) or a full screen window should be good.

**Comment by officert** [comment]
@jojobyte sounds good, just created the issue here.

https://github.com/officert/mongotron/issues/61

**Comment by hasantayyar** [comment]
As mentioned below ssh tunneling will be a great feature.
Also in text view mode mongontron can highlight the json texts.

**Comment by beeman** [comment]
Hey @officert, thanks a lot for this publishing this tool!

I would like to see an option to clone collections. I needed this yesterday and found out that there are very few tools that support this.

**Comment by officert** [comment]
Small releases. Right now the OSX release is around 130mb, which is huggggge. Sounds like around that size is expected

https://github.com/atom/electron/issues/2003

but looking at the Slack app on OSX it's only 18mb so I think with some fine tuning we can bring down the size quite a bit. 

Need to look into excluding any unnecessary node modules.

**Comment by YerkoPalma** [comment]
There is a similar GUI tool for Mongo, called MongoVue. It is not open source, and have some interesting [features](http://www.mongovue.com/features/). What I think it would be awesome, is the feature to convert from standar SQL syntax to MongoDB. 
I don't know how this could be done, so this is just a suggestion, I guess that it will be needed some kind of parser or something like that. MongoDB has some [sql comparisson](https://docs.mongodb.org/v3.0/reference/sql-comparison/) on the docs, and there are some [online converters](http://www.querymongo.com/) too.
And finally, there is a [npm package](https://www.npmjs.com/package/mongo-sql) that might be interesting (just saw it now, so I haven't tested it) and also a [gem](http://mongify.com/) that do something similar. I don't know what you think about this kind of feature

**Comment by officert** [comment]
@jojobyte I've been looking into SSH tunnel support and I'm not sure that we need it. Since we're using the node mongo driver to connect to database instances directly, and not using the mongo shell like other projects (Robomongo), do we need SSH? Do you mind describing your use case?

**Comment by jojobyte** [comment]
@officert I believe those are two separate situations. So far as I understand, SSH Tunneling would be used when you want to connect to a mongodb that is not open to the internet, IE. mongos config has `bind_ip = 127.0.0.1` or something similar where you can only connect from that server. The tunneling would essentially SSH into the server using your credentials then associate a port on your local machine with the remote servers mongo port which allows you to connect without specifically opening mongo to the internet.

Useful if you're running mongo on the same server as your Node.js instance or something similar.

Not sure if there are any security concerns or implications to take into account with this, but I've seen the feature in other GUI DB managers before.

**Comment by officert** [comment]
@jojobyte thanks yea I see what you're saying now. I have also seen this in other DB managers. I'll leave this issue open https://github.com/officert/mongotron/issues/56. I've started to look at it a bit, looks like there are a handful of node ssh libs out there we could use.

**Comment by jelyman2** [comment]
@officert making sure you're respecting symlinks and `zip`ping will help. Gotta remember a large part of Electron is Chrome.

**Comment by xeo-it** [comment]
I'm also "voting" for ssh tunnel support, in my company we access our databases only via ssh for the same reason @jojobyte described above.  
Thanks a lot for developing this awesome tool :+1:

**Comment by simonbengtsson** [comment]
Have it been any discussion on a table view of the collections? Similar to how mongo-express does it.

**Comment by officert** [comment]
@simonbengtsson there haven't been any discussions about adding a table view, but I think it's a great idea! If it's something you're interested in working on please go ahead and create a new issue for it. Thanks!

**Comment by simonbengtsson** [comment]
Sweet! Will do!
